notebooks/ch17/classification_sensitivity_specificity.md

```diff
@@ -0,0 +1,486 @@
+
+
+```python
+spam_only_cnf_matrix = confusion_matrix(y_train, spam_only_y_pred, labels=[1, 0])
+
+plot_confusion_matrix(spam_only_cnf_matrix, classes=class_names,
+                      title='spam_only Confusion Matrix')
+```
+
+
+![png](classification_sensitivity_specificity_files/classification_sensitivity_specificity_15_0.png)
+
+
+At the other extreme, `spam_only` predicts the training dataset has no ham emails, which the confusion matrix indicates is far from the truth with 758 false positives.
+
+Our main interest is in the confusion matrix for `words_list_model`:
+
+
+```python
+words_list_model_cnf_matrix = confusion_matrix(y_train, words_list_model_y_pred, labels=[1, 0])
+
+plot_confusion_matrix(words_list_model_cnf_matrix, classes=class_names,
+                      title='words_list_model Confusion Matrix')
+```
+
+
+![png](classification_sensitivity_specificity_files/classification_sensitivity_specificity_17_0.png)
+
+
+The row totals match those of the `ham_only` and `spam_only` confusion matrices as expected since the true labels in the training dataset are unaltered for all models.
+
+Of the 42 spam emails, `words_list_model` correctly classifies 18 of them, which is a poor performance. Its high accuracy is buoyed by the large number of true negatives, but this is insufficent because it does not reliably filter out spam emails.
+
+This emails dataset is an example of a **class-imbalanced dataset**, in which a vast majority of labels are in one class over the other. In this case, most of our emails are ham. Another common example of class imbalance is disease detection when the frequency of the disease in a population is low. A medical test that always concludes a patient doesn't have the disease will have a high accuracy, but its inability to identify individuals with the disease renders it useless. 
+
+We now turn to sensitivity and specificity, two metrics that are better suited for evaluating class-imbalanced datasets.
+
+### Sensitivity
+
+**Sensitivity** (also called **true positive rate**) measures the proportion of data belonging to the positive class that the classifier correctly labels.
+
+$$ \text{Sensitivity} = \frac{TP}{TP + FN} $$
+
+From our discussion of confusion matrices, you should recognize the expression $TP + FN$ as the sum of the entries in the first row. Building confusion matrices allows us to easily compare the sensitivities of our models:
+
+* `ham_only`: $\frac{0}{0 + 42} = 0$
+* `spam_only`: $\frac{42}{42 + 0} = 1$
+* `words_list_model`: $\frac{18}{18 + 24} \approx .429$
+
+Since `ham_only` has no true positives, it has the worst possible sensitivity value of 0. On the other hand, `spam_only` has an abysmally low accuracy but it has the best possible sensitivity value of 1. The low sensitivity of `words_list_model` indicates that it frequently fails to mark spam emails as such. Nonetheless, it outperforms `ham_only`, which will prove significant when we examine ROC curves later.
```

Comment:
I don't think we need to mention "will prove significant when we examine ROC curves later." since ROC curves have no meaning for the reader right now and the importance of sensitivity will be clear when we get to the curve.